front/src/applications/stdcm/hooks/useStdcm.ts

```diff
@@ -87,74 +93,260 @@ const useStdcm = ({
     }
   };
 
+  const getPayloadWithTolerance = (payload: PostTimetableByIdStdcmApiArg, tolerance: number) => ({
+    ...payload,
+    body: {
+      ...payload.body,
+      steps: payload.body.steps.map((step) => modifyStepTolerance(step, tolerance)),
+    },
+  });
+
+  const createSimulation = async (
+    index: number,
+    inputs: StdcmSimulationInputs,
+    response: StdcmResponse,
+    stdcmTrain: TrainScheduleResult | undefined,
+    { alternativePath = false, upstreamPath = false } = {}
+  ): Promise<StdcmSimulation> => {
+    const pathProperties = await fetchPathProperties(response.path, infraId, dispatch);
+    if (!pathProperties || !stdcmTrain) {
+      throw new Error('Path properties could not be fetched.');
+    }
+    let outputs;
+    if (response.status === 'success') {
+      // Compute chart data when successful.
+      const chartData = computeChartData(
+        response,
+        stdcmTrain,
+        t,
+        stdcmRollingStock as RollingStockWithLiveries,
+        pathProperties
+      );
+      outputs = {
+        pathProperties,
+        results: response,
+        speedSpaceChartData: chartData,
+      };
+    } else {
+      outputs = { pathProperties, conflicts: response.conflicts ?? [] };
+    }
+    return {
+      index,
+      creationDate: new Date(),
+      inputs,
+      outputs,
+      alternativePath,
+      upstreamPath,
+    };
+  };
+
+  const updateTrainStartTime = (
+    train: TrainScheduleResult,
+    response: StdcmResponse
+  ): TrainScheduleResult | undefined =>
+    response.status === 'success' ? { ...train, start_time: response.departure_time } : undefined;
+
+  const handleSuccess = async (
+    response: StdcmApiSuccessResponse,
+    payload: PostTimetableByIdStdcmApiArg
+  ) => {
+    setCurrentStdcmRequestStatus(STDCM_REQUEST_STATUS.success);
+
+    const successResponse = {
+      ...response,
+      rollingStock: stdcmRollingStock,
+      creationDate: new Date(),
+      speedLimitByTag: osrdconf.speedLimitByTag,
+      simulationPathSteps: osrdconf.stdcmPathSteps,
+    } as StdcmSuccessResponse;
+
+    const stdcmTrain: TrainScheduleResult = {
+      id: STDCM_TRAIN_ID,
+      timetable_id: timetableId,
+      comfort: payload.body.comfort,
+      constraint_distribution: 'MARECO',
+      path: payload.body.steps.map((step) => ({ ...step.location, id: nextId() })),
+      rolling_stock_name: stdcmRollingStock!.name,
+      start_time: response.departure_time,
+      train_name: 'stdcm',
+    };
+    dispatch(updateSelectedTrainId(STDCM_TRAIN_TIMETABLE_ID));
+
+    const simulation = await createSimulation(
+      simulationsList.length,
+      currentSimulationInputs,
+      successResponse,
+      stdcmTrain
+    );
```

Comment:
You can use the function createSimulation here
```suggestion
  const simulaiton = await createSimulation(...)
```

Reply:
This simplified the code even more when adding the formatting to `createSimulation`